src/Compilers/CSharp/Portable/Generated/Syntax.xml.Internal.Generated.cs

```diff
@@ -27085,51 +27085,70 @@ static PropertyDeclarationSyntax()
   /// <summary>The syntax for the expression body of an expression-bodied member.</summary>
   internal sealed partial class ArrowExpressionClauseSyntax : CSharpSyntaxNode
   {
+    internal readonly SyntaxToken readOnlyKeyword;
     internal readonly SyntaxToken arrowToken;
     internal readonly ExpressionSyntax expression;
 
-    internal ArrowExpressionClauseSyntax(SyntaxKind kind, SyntaxToken arrowToken, ExpressionSyntax expression, DiagnosticInfo[] diagnostics, SyntaxAnnotation[] annotations)
+    internal ArrowExpressionClauseSyntax(SyntaxKind kind, SyntaxToken readOnlyKeyword, SyntaxToken arrowToken, ExpressionSyntax expression, DiagnosticInfo[] diagnostics, SyntaxAnnotation[] annotations)
```

Comment:
This is definitely a WAT to see.  Why would keywords be attached here, as opposed to the actual member which could be readonly?

Reply:
It's to support the syntax `public int Property readonly => 42;`, which means the same thing as `public int Property { readonly get => 42; }`.

Feel free to weigh in on the concrete syntax design in dotnet/csharplang#1744.

Reply:
yeah, i don't see how it make any sense for that to be on the arrow-expression as opposed to be on the property decl.

Reply:
> Feel free to weigh in on the concrete syntax design in [dotnet/csharplang#1710](https://github.com/dotnet/csharplang/issues/1710). (edit: whoops, had the wrong issue ID)

Sorry, where in that thread is a discussion of the Roslyn syntax impl?

Reply:
To be clear: i'm not saying it's wrong to allow the user to write: `public int Property readonly => 42;`  I get that that's part of the *language feature*.  What i'm saying is: i would not *implement* this by havin the `readonly` keyword belong to the ArrowExpression node.  It should belong to hte PropertyDecl node.

Reply:
Fair enough--would you do it by adding a ReadOnlyKeyword property to PropertyDeclarationSyntax or by appending the readonly token to Modifiers?

Reply:
My thinking was to try and convey "either this property has an expression body with an optional `readonly`, or it has an accessor list". But I see how that could cause confusion in other parts of Roslyn that use ArrowExpressionClause.

Reply:
> Fair enough--would you do it by adding a ReadOnlyKeyword property to PropertyDeclarationSyntax

Yes.

> or by appending the readonly token to Modifiers?

This coudl not work.  The modifiers show up on different side of the declaration.  Syntax, is 'concrete', and thus is represented in memory exactly as it was in source.  So it's not possible for something like 'modifiers' to include data from both sides of another construct.

Reply:
> My thinking was to try and convey "either this property has an expression body with an optional `readonly`, or it has an accessor list". But I see how that could cause confusion in other parts of Roslyn that use ArrowExpressionClause.

Even if that was the case, i would be extremely surprised if one could not write `readonly accessor_list`.  Why would we force `int Prop readonly => ...` and not allow `int Prop readonly { get => ... }`?

That would be a pretty bizarre asymmetry in the language.

Also, are you supporting readonly methods? If so, we'd clearly need a place to put the readonly that was not on the arrow-expression.

Reply:
BAsed on the csharplang issue you linked to, the following would be legal:

```c#
    public float GetLengthReadonly() readonly
    {
        return MathF.Sqrt(LengthSquared);
    }
```

So methods would clearly need to have a place to put 'readonly'.  And it would definitely be bad to put `readonly` on `block_syntax`.

--

Note: you should also not add `readonly` as a fixed keyword here. Instead, you should have `trailing modifiers`.  For all we know, we'll add more modifiers in the future, and it's unlikely that we'd want to force a specific order here.

Reply:
@CyrusNajmabadi The issue is out of date, LDM determined it should be placed in the standard modifier list: 
https://github.com/dotnet/csharplang/issues/1710#issuecomment-429992790

Reply:
Ah ok.  That's super simple then.  No need to do anything since hte parser will already consume readonly in the modifiers list for any memeber.  Hooray :)

Reply:
I've updated the speclet based on recent language design discussions. Unless new feedback or issues arise in the next day or so, I'll move forward with the newer design, which involves removing the `ReadOnlyKeyword` property and using the `readonly int Prop => ...` form.